browser-test/src/checkbox.test.ts

```diff
@@ -0,0 +1,204 @@
+    pageObject = page
+  })
+
+  afterEach(async () => {
+    await resetSession(pageObject)
+  })
+
+  describe('single checkbox question', () => {
+    let applicantQuestions
+    const programName = 'test program for single checkbox'
+
+    beforeAll(async () => {
+      // As admin, create program with single checkbox question.
+      await loginAsAdmin(pageObject)
+      const adminQuestions = new AdminQuestions(pageObject)
+      const adminPrograms = new AdminPrograms(pageObject)
+      applicantQuestions = new ApplicantQuestions(pageObject)
+
+      await adminQuestions.addCheckboxQuestion({
+        questionName: 'checkbox-color-q',
+        options: ['red', 'green', 'orange', 'blue'],
+        minNumChoices: 1,
+        maxNumChoices: 2,
+      })
+      await adminPrograms.addAndPublishProgramWithQuestions(
+        ['checkbox-color-q'],
+        programName,
+      )
+
+      await logout(pageObject)
+    })
+
+    it('with single checked box submits successfully', async () => {
+      await loginAsGuest(pageObject)
+      await selectApplicantLanguage(pageObject, 'English')
+
+      await applicantQuestions.applyProgram(programName)
+      await applicantQuestions.answerCheckboxQuestion(['blue'])
+      await applicantQuestions.clickNext()
+
+      await applicantQuestions.submitFromReviewPage(programName)
+    })
+
+    it('with no checked boxes does not submit', async () => {
+      await loginAsGuest(pageObject)
+      await selectApplicantLanguage(pageObject, 'English')
+
+      await applicantQuestions.applyProgram(programName)
+
+      // No validation errors on first page load.
+      const checkBoxError = '.cf-applicant-question-errors'
+      expect(await pageObject.isHidden(checkBoxError)).toEqual(true)
+
+      // Click next without selecting anything.
+      await applicantQuestions.clickNext()
+
+      // Check checkbox error and required error are present.
+      expect(await pageObject.isHidden(checkBoxError)).toEqual(false)
+      const checkboxId = '.cf-question-checkbox'
+      expect(await pageObject.innerText(checkboxId)).toContain(
+        'This question is required.',
+      )
+    })
+
+    it('with greater than max allowed checked boxes does not submit', async () => {
+      await loginAsGuest(pageObject)
+      await selectApplicantLanguage(pageObject, 'English')
+
+      await applicantQuestions.applyProgram(programName)
+      const checkBoxError = '.cf-applicant-question-errors'
+      // No validation errors on first page load.
+      expect(await pageObject.isHidden(checkBoxError)).toEqual(true)
+
+      // Max of two checked boxes are allowed, but we select three.
+      await applicantQuestions.answerCheckboxQuestion([
+        'blue',
+        'green',
+        'orange',
+      ])
+      await applicantQuestions.clickNext()
+
+      // Check error is shown.
+      expect(await pageObject.isHidden(checkBoxError)).toEqual(false)
+    })
+  })
+
+  describe('multiple checkbox questions', () => {
```

Comment:
Actually, I feel like this type of test should not belong in the question type specific browser tests. This feels independent of what question type is being used. Instead, I'd like to create just one of these tests in application_navigation.test.ts, and remove it from all the question type specific tests. WDYT?

Reply:
I believe we have these because FE validation is bespoke for each type; at one point some questions handled multiples of their types and others did not.